packages/salesforcedx-vscode-apex-debugger/src/core/streamingService.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  CommandExecution,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { CommandOutput } from '../utils/commandOutput';
+import { StreamingClient, StreamingClientInfo } from './streamingClient';
+
+export interface OrgInfo {
+  username: string;
+  devHubId: string;
+  id: string;
+  createdBy: string;
+  createdDate: string;
+  expirationDate: string;
+  status: string;
+  edition: string;
+  orgName: string;
+  accessToken: string;
+  instanceUrl: string;
+  clientId: string;
+}
+
+export class StreamingService {
+  public static SYSTEM_EVENT_CHANNEL = '/systemTopic/ApexDebuggerSystemEvent';
+  public static USER_EVENT_CHANNEL = '/systemTopic/ApexDebuggerEvent';
+  public static DEFAULT_TIMEOUT = 14400;
+  private static instance: StreamingService;
+  private clients: StreamingClient[] = [];
+
+  public static getInstance() {
+    if (!StreamingService.instance) {
+      StreamingService.instance = new StreamingService();
+    }
+    return StreamingService.instance;
+  }
+
+  public getClients(): StreamingClient[] {
+    return this.clients;
+  }
+
+  public async subscribe(
+    project: string,
```

Comment:
I see that the signature of getOrgInfo actually wants a projectDir.

Is it better to explicitly call this parameter `projectPath` rather than `project` just to make it easier?
That way, it is also clear that this is a string that can contain / or \ depending on the OS.